backend/taxHandlers.ts

```diff
@@ -0,0 +1,102 @@
+import { check } from "prettier";
+import { TaxForOrderRes } from "taxjar/dist/util/types";
+import { OrderSubscriptionFragment } from "../generated/graphql";
+import { createOrderTransaction, fetchTaxes } from "./taxjarApi";
+import {
+  CheckoutPayload,
+  ResponseTaxPayload,
+  FetchTaxesPayload,
+  TaxJarConfig,
+  OrderPayload,
+} from "./types";
+
+const _calculateTaxes = async (
+  taxData: FetchTaxesPayload,
+  taxJarConfig: TaxJarConfig
+): Promise<{ data: ResponseTaxPayload }> => {
+  const taxResposne = await fetchTaxes(taxData, taxJarConfig);
+  const taxDetails = taxResposne.tax.breakdown;
+  const shippingDetails = taxDetails?.shipping;
+  let shippingPriceGross = Number(taxData.shipping_amount);
+  let shippingPriceNet = Number(taxData.shipping_amount);
+  let shippingTaxRate = 0;
+
+  if (taxDetails !== undefined) {
+    if (shippingDetails !== undefined) {
+      shippingPriceGross =
+        shippingDetails.taxable_amount + shippingDetails.tax_collectable;
+      shippingPriceNet = shippingDetails.taxable_amount;
+      shippingTaxRate = shippingDetails.combined_tax_rate;
+    }
+  }
+
+  return {
+    data: {
+      shipping_price_gross_amount: String(shippingPriceGross),
+      shipping_price_net_amount: String(shippingPriceNet),
+      shipping_tax_rate: String(shippingTaxRate),
+      // lines order needs to be the same as for recieved payload.
+      lines: taxData.lines.map((line) => {
+        let totalGrossAmount = line.total_amount;
+        let totalNetAmount = line.total_amount;
+        let taxRate = "0";
+        if (taxDetails !== undefined && taxDetails.line_items !== undefined) {
+          const lineTax = taxDetails.line_items.find((l) => l.id === line.id);
+          if (lineTax !== undefined) {
+            totalGrossAmount = String(
+              lineTax.taxable_amount + lineTax.tax_collectable
+            );
+            totalNetAmount = String(lineTax.taxable_amount);
+            taxRate = String(lineTax.combined_tax_rate);
+          }
+        }
+        return {
+          total_gross_amount: totalGrossAmount,
+          total_net_amount: totalNetAmount,
+          tax_rate: taxRate,
+        };
+      }),
+    },
+  };
+};
+
+export const calculateCheckoutTaxes = async (
+  checkoutPayload: CheckoutPayload,
+  taxJarConfig: TaxJarConfig
+): Promise<{ data: ResponseTaxPayload }> => {
+  const taxData: FetchTaxesPayload = {
+    address: checkoutPayload.address,
+    lines: checkoutPayload.lines,
+    channel: checkoutPayload.channel,
+    shipping_amount: checkoutPayload.shipping_amount,
+    shipping_name: checkoutPayload.shipping_name,
+  };
```

Comment:
@jwm0  `FetchTaxesPayload` and `calculateTaxes` is shared between sync webhook responsible for calculating taxes for checkout and order.  The webhooks have a different payloads, so I decided that it would be good to have a type that will store only a tax-important data for us. 
What do you propose? Should I create a common part of declared interface and inherit it in declaration of checkout payload and order payload?